modules/discoveryBidAdapter.js

```diff
@@ -0,0 +1,484 @@
+    for (let bid of bids) {
+      let impid = getKv(bid, 'impid');
+      if (itemMaps[impid]) {
+        let bidId = getKv(itemMaps[impid], 'req', 'bidId');
+        const mediaType = getKv(bid, 'w') ? 'banner' : 'native';
+        let bidResponse = {};
+        if (mediaType === 'native') {
+          const adm = getKv(bid, 'adm');
+          const admObj = JSON.parse(adm);
+          var native = {};
+          admObj.assets.forEach((asset) => {
+            if (asset.title) {
+              native.title = asset.title.text;
+            } else if (asset.data) {
+              native.data = asset.data.value;
+            } else if (asset.img) {
+              switch (asset.img.type) {
+                case 1:
+                  native.icon = {
+                    url: asset.img.url,
+                    width: asset.img.w,
+                    height: asset.img.h,
+                  };
+                  break;
+                default:
+                  native.image = {
+                    url: asset.img.url,
+                    width: asset.img.w,
+                    height: asset.img.h,
+                  };
+                  break;
+              }
+            }
+          });
+          if (admObj.link) {
+            if (admObj.link.url) {
+              native.clickUrl = admObj.link.url;
+            }
+          }
+          if (Array.isArray(admObj.eventtrackers)) {
+            native.impressionTrackers = [];
+            admObj.eventtrackers.forEach((tracker) => {
+              if (tracker.event !== 1) {
+                return;
+              }
+              switch (tracker.method) {
+                case 1:
+                  native.impressionTrackers.push(tracker.url);
+                  break;
+                // case 2:
+                //   native.javascriptTrackers = `<script src=\"${tracker.url}\"></script>`;
+                //   break;
+              }
+            });
+          }
+          if (admObj.purl) {
+            native.purl = admObj.purl;
+          }
+          bidResponse = {
+            requestId: bidId,
+            cpm: getKv(bid, 'price'),
+            creativeId: getKv(bid, 'cid'),
+            mediaType,
+            currency: cur,
+            netRevenue: true,
+            native,
+            ttl: TIME_TO_LIVE,
+            nurl: getKv(bid, 'nurl'),
+            meta: {
+              advertiserDomains: getKv(bid, 'adomain') || []
+            }
+          };
+        } else {
+          bidResponse = {
+            requestId: bidId,
+            cpm: getKv(bid, 'price'),
+            width: getKv(bid, 'w'),
+            height: getKv(bid, 'h'),
+            creativeId: getKv(bid, 'cid'),
+            dealId: '',
+            mediaType,
+            currency: cur,
+            netRevenue: true,
+            ttl: TIME_TO_LIVE,
+            ad: getKv(bid, 'adm'),
+            nurl: getKv(bid, 'nurl'),
+            meta: {
+              advertiserDomains:  getKv(bid, 'adomain') || []
```

Comment:
Sorry one last thing here. I think you can setup the shared bidResponse object properties when declaring the bidResponse object on line 372, then add the unique mediatype props in the conditional to remove repetition here

Reply:
OK, it has been modified, because I want to merge into master as soon as possible, so I always forget to check myself and give you trouble.